Src/FluentAssertions/Common/MemberPath.cs

```
@@ -21,23 +26,37 @@ public bool IsParentOrChildOf(string candidate)
             return IsParent(candidate) || IsChild(candidate);
         }
 
+        public bool IsSameAs(string candidate, Type memberDeclaringType)
+        {
+            string[] candidateSegments = Segmentize(candidate);
```

🙃 fail fast on `memberDeclaringType == declaringType`
I.e. `if (memberDeclaringType != declaringType) return false;`